.github/workflows/pytest.yml

```diff
@@ -0,0 +1,46 @@
+name: Pytest
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install pytest pytest-retry
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        echo "PYTHONPATH=." >> "$GITHUB_ENV"
+
+    - name: Running tests for the rest code samples
+      # continue-on-error: true
+      env:
```

Comment:
Consider having a `.env` file, not for this PR for another one

Reply:
I'm making use of GitHub secrets and exporting them as environment variables for later commands to be aware of. They're secrets so they shouldn't be visible. If there is a way to use a .`env` file without exposing such secrets please let me know. https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions